dataflux_pytorch/dataflux_iterable_dataset.py

```diff
@@ -136,11 +135,18 @@ def __init__(
 
         # If composed download is enabled, check if the client has permissions to create and delete the composed object.
         if self.config.max_composite_object_size != 0:
-            missing_perm = _get_missing_permissions(
-                storage_client=self.storage_client,
-                bucket_name=self.bucket_name,
-                project_name=self.project_name,
-                required_perm=[CREATE, DELETE])
+            try:
+                missing_perm = _get_missing_permissions(
+                    storage_client=self.storage_client,
+                    bucket_name=self.bucket_name,
+                    project_name=self.project_name,
+                    required_perm=[CREATE, DELETE])
+            except RefreshError as e:
+                e.add_note(
+                    "Default credentials might be missing. Run `gcloud auth application-default login`")
```

Comment:
Maybe this should point to the GCP docs page on configuring app-default credentials instead? Running the gcloud command would be the right advice when running locally, but won't help users who are on GKE for example.